conans/test/integration/toolchains/google/test_bazeldeps.py

```diff
@@ -880,3 +880,84 @@ def build(self):
           "--build=example")
     c.assert_listed_require({"example/1.0": "Cache"})
     c.assert_listed_require({"example/1.0": "Cache"}, build=True)
+
+
+class TestBazelGenerationBuildContext:
+    """
+    https://github.com/conan-io/conan/issues/15764
+    """
+    def test_bazel_generate(self):
+        c = TestClient()
+        tool = textwrap.dedent("""
+            import os
+            from conan import ConanFile
+            from conan.tools.google import BazelDeps
+            class Example(ConanFile):
+                name = "tool"
+                version = "1.0"
+                requires = "wayland/1.0"
+                tool_requires = "wayland/1.0"
+                def generate(self):
+                    tc = BazelDeps(self)
+                    tc.build_context_activated = ["wayland", "dep"]
+                    tc.generate()
+                def build(self):
+                    context = "build-" if self.context == "build" else ""
```

Comment:
@memsharded The context is always `build` (just checked putting an assert in this build step). Is that expected?